internal/provider/app_entitlement_resource.go

```diff
@@ -413,6 +426,31 @@ func (r *AppEntitlementResource) Update(ctx context.Context, req resource.Update
 		appEntitlement.ProvisionPolicy = currentAppEntitlement.ProvisionPolicy
 	}
 
+	// TODO(mstanbCO): Need a better pattern for handling this in the merge step, instead of doing this.
+	if currentAppEntitlement.ProvisionPolicy != nil {
+		if currentAppEntitlement.ProvisionPolicy.ConnectorProvision != nil {
+			if appEntitlement.ProvisionPolicy.DelegatedProvision != nil || appEntitlement.ProvisionPolicy.ManualProvision != nil ||
```

Comment:
can we make the appEntitlement.ProvisionPolicy.DelegatedProvision != nil  into a var like 

> isDelegatedSet := appEntitlement.ProvisionPolicy.DelegatedProvision != nil 
> ...
> isDelegatedSet || isManualSet || isWebhookSet

I think that would make it a lot cleaner